code/modules/projectiles/gun.dm

```diff
@@ -200,13 +202,15 @@
 		else
 			dual_wielding = FALSE
 
-		if(off_hand && off_hand.can_hit(user))
+		if(!can_dual)
+			dual_wielding = FALSE
+		else if(off_hand && off_hand.can_hit(user))
 			spawn(1)
 			off_hand.Fire(A,user,params)
-	else
-		dual_wielding = FALSE
+		else
+			dual_wielding = FALSE
 
-	Fire(A,user,params) //Otherwise, fire normally.
+		Fire(A,user,params) //Otherwise, fire normally.
```

Comment:
This should not be indented. It makes guns impossible to fire on any intent but harm.

Reply:
fixed!